src/main/java/com/avispl/symphony/dal/infrastructure/management/tripleplay/TriplePlayAggregatorCommunicator.java

```diff
@@ -0,0 +1,601 @@
+import com.avispl.symphony.dal.util.StringUtils;
+import com.avispl.symphony.dal.util.ControllablePropertyFactory;
+
+/**
+ * Communicator
+ *
+ * @author Harry / Symphony Dev Team<br>
+ * Created on 3/13/2023
+ * @since 1.0.0
+ */
+public class TriplePlayAggregatorCommunicator extends RestCommunicator implements Aggregator, Monitorable, Controller {
+
+	class ClientLoader implements Runnable {
+		private volatile List<String> clientIps;
+
+		public ClientLoader(List<String> clientIps) {
+			this.clientIps = clientIps;
+		}
+
+		@Override
+		public void run() {
+
+			if (!cachedClients.isEmpty()) {
+				retrieveClient(this.clientIps);
+			}
+		}
+	}
+
+	/**
+	 * ReentrantLock to prevent null pointer exception to localExtendedStatistics when controlProperty method is called before GetMultipleStatistics method.
+	 */
+	private final ReentrantLock reentrantLock = new ReentrantLock();
+
+	/**
+	 * Executor that runs all the async operations
+	 */
+	private static ExecutorService executorService;
+
+	private volatile int localPollingInterval = TriplePlayConstrant.MIN_POLLING_INTERVAL;
+	private volatile int deviceStatisticsCollectionThreads;
+
+	/**
+	 * store pollingInterval adapter properties
+	 */
+	private volatile String pollingInterval;
+
+	private volatile String lastClientMac;
+	private List<Future> clientExecutionPool = new ArrayList<>();
+	private TreeMap<String, Client> cachedClients = new TreeMap<>();
+	private Map<String, AggregatedDevice> cachedAggregatedDevices = new ConcurrentHashMap<>();
+	private HashMap<String, Service> cachedServices = new HashMap<>();
+	private boolean isEmergencyDelivery = false;
+	private ObjectMapper objectMapper = new ObjectMapper();
+
+	@Override
+	public List<Statistics> getMultipleStatistics() throws Exception {
+		if (executorService != null) {
+			for (Future future : clientExecutionPool) {
+				future.cancel(true);
+			}
+		}
+		reentrantLock.lock();
+		try {
+			if (!isEmergencyDelivery) {
+				int currentSizeCacheClients = cachedClients.size();
+				retrieveClients();
+				retrieveService();
+				localPollingInterval = calculatingLocalPollingInterval();
+				deviceStatisticsCollectionThreads = calculatingThreadQuantity();
+				//Multi thread for get information of client
+				retrieveInformationOfAllClients(currentSizeCacheClients);
+			}
+			isEmergencyDelivery = false;
+		} finally {
+			reentrantLock.unlock();
+		}
+		return null;
+	}
+
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+		String property = controllableProperty.getProperty();
+		String value = String.valueOf(controllableProperty.getValue());
+		String deviceId = controllableProperty.getDeviceId();
+		reentrantLock.lock();
+		try {
+			if (StringUtils.isNullOrEmpty(deviceId) || !cachedAggregatedDevices.containsKey(deviceId)) {
+				return;
+			}
+			aggregatedDeviceControl(property, value, cachedAggregatedDevices.get(deviceId));
```

Comment:
We need to verify if the control is exact from aggregatedDeviceControl or not. Maybe the code can run now, but in the future if we want to scale up to support more function such as Aggregator control it will cause some problem